### PR TITLE
DDP-4876 add requirement flags to address component

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Add `requireVerified` and `requirePhone` flags to address component. Older clients that doesn't understand these flags can ignore them (and so as if it's `false`). I am NOT able to implement checking these requirements on the backend because we're missing address data.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/docs/specification/src/pepper.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ openapi: 3.0.0
 servers: []
 info:
   description: 'Pepper API specification'
-  version: 1.8.0
+  version: 1.9.0
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/pepper.yml:5]: Should be `v1.9` now.

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 2]
```diff
@@ -686,6 +686,12 @@ Component.MailingAddress:
               type: string
             subtitleText:
               type: string
+            requireVerified:
+              type: boolean
+              description: when `true`, address should be verified as deliverable
+            requirePhone:
+              type: boolean
+              description: when `true`, phone field should be filled out
```
Comment (yufengwng) [inline pepper-apis/docs/specification/src/components/schemas.yml]: CC @bskinner @fat23cat

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/MailingAddressComponent.java [hunk 3]
```diff
@@ -16,12 +16,15 @@
...
 
-    public MailingAddressComponent(Long titleTemplateId, Long subtitleTemplateId, boolean hideNumber) {
+    public MailingAddressComponent(Long titleTemplateId, Long subtitleTemplateId, boolean hideNumber,
+                                   boolean requireVerified, boolean requirePhone) {
         super(ComponentType.MAILING_ADDRESS);
         hideDisplayNumber = hideNumber;
-        titleTemplateId = titleTemplateId;
-        subtitleTemplateId = subtitleTemplateId;
+        this.titleTemplateId = titleTemplateId;
+        this.subtitleTemplateId = subtitleTemplateId;
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/MailingAddressComponent.java:24]: This was a bug! We got overshadowed variables here and need to use `this` qualifier! This resulted in title/subtitle NOT being assigned properly.

File: study-builder/studies/testboston/address.conf [hunk 4]
```diff
@@ -39,6 +39,8 @@
             ]
           },
           "subtitleTemplate": null,
+          "requireVerified": true,
+          "requirePhone": true,
```
Comment (yufengwng) [inline study-builder/studies/testboston/address.conf:30]: TestBoston will require both.

File: study-builder/studies/testboston/study-events.conf [hunk 5]
```diff
@@ -100,6 +100,32 @@
...
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": ${id.act.address},
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "ANNOUNCEMENT",
+        "msgTemplate": {
+          "templateType": "HTML",
+          "templateText": """<p class="infobox-text_small">$tb_done_msg_p1</p>""",
```
Comment (yufengwng) [inline study-builder/studies/testboston/study-events.conf]: @fat23cat is this the CSS we should use?

File: study-builder/studies/testboston/study-events.conf [hunk 6]
```diff
@@ -100,6 +100,32 @@
     },
 
     # when completing address
+    {
```
Comment (yufengwng) [inline study-builder/studies/testboston/study-events.conf]: For https://broadinstitute.atlassian.net/browse/DDP-4876.

File: study-builder/studies/testboston/study-events.conf [hunk 7]
```diff
@@ -112,9 +138,37 @@
...
       "maxOccurrencesPerUser": 1,
-      "delaySeconds" : 259200 #72hrs
+      "delaySeconds" : 259200, # 72hrs
       "dispatchToHousekeeping": true,
-      "order": 4
+      "order": 2
+    },
+
+    # when completing baseline symptoms
+    {
```
Comment (yufengwng) [inline study-builder/studies/testboston/study-events.conf]: For https://broadinstitute.atlassian.net/browse/DDP-4880.

File: pepper-apis/docs/specification/src/components/schemas.yml (hunk 2 above)
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml]:
```suggestion
              description: |
                     when `true`, the activity may require that there be a saved mailing address with a non-empty value for `phone` in order to be [completed](#operation/putAnswers).
```

File: pepper-apis/docs/specification/src/components/schemas.yml [hunk 8]
```diff
@@ -686,6 +686,12 @@ Component.MailingAddress:
               type: string
             subtitleText:
               type: string
+            requireVerified:
+              type: boolean
+              description: when `true`, address should be verified as deliverable
```
Comment (bskinner) [inline pepper-apis/docs/specification/src/components/schemas.yml]:
```suggestion
              description: when `true`, address should be [verified](#operation/verifyMailingAddress) as deliverable
```

**Review by bskinner — COMMENTED** [review]
Made a few suggestions on the documentation, but looks good otherwise. Nice find with the shadowing bug!

**Review by MocanaAtBroad — DISMISSED** [review]
We will have to modify client to ensure that saving of address completes before PUT to have this work properly

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**